### DISCUSSION TITLE
锤子便签辅助增强

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2719] [插件说明]
描述
---

> 为锤子便签提供辅助性增强功能。

地址
---

[https://simpread.ksria.cn/plugins/details/QSRqB3UAIy](https://simpread.ksria.cn/plugins/details/QSRqB3UAIy)

功能
---

> 支持 `YAML` 描述

![image](https://user-images.githubusercontent.com/81074/130903862-cf7e7cda-c6ce-4254-890a-efefde8e840d.png)

### 可识别的  `YAML` 语法

> `title`

当设定为 title 时，在进入阅读模式后会自动设置标题为 `title` 内容

![image](https://user-images.githubusercontent.com/81074/130904059-61b57043-973e-48db-a732-2397ba5ca386.png)

如何使用
---

> 细节 [请看这里](https://github.com/Kenshin/simpread/discussions/2708)。

选项
---

![image](https://user-images.githubusercontent.com/81074/130904683-9f9d7567-68a2-41df-b96c-24931ac572db.png)

更新日志
---

### 1.0.1

- `Optimize` 为配合 [插件管理器](https://github.com/Kenshin/simpread/discussions/4221) 的更新方式做了数据结构的优化。

### 1.0.0

- 修复 模仿 Safari 阅读模式样式 在锤子便签中的样式错误
- 读取 `YAML`
    - 当存在 title 时进入阅读模式后是否设置为标题

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by Kenshin** [comment]
YAML 是专门用来写配置文件的语言，非常简洁和强大，远比 JSON 格式方便，更多介绍 [请看这里](https://www.ruanyifeng.com/blog/2016/07/yaml.html)。

> 目前在锤子便签中，仅支持 `title` 的识别（如何使用请看上方教程），如果你有其它需要识别的字段，请在这里给我留言。

**Reply by YB-Kang** [reply]
我点击简悦图标老是智能感知失败，没法用

**Reply by Kenshin** [reply]
应该是没问题，我几乎每几天都会测试下。

建议录屏或截图详细说明下，另外你可以刷新页面后再试试。

**Reply by Kenshin** [reply]
检查下：

1. 锤子插件能否显示选项页
2. Markdown 转换 HTML能否显示选项页

你给出的截图根本就没有截图出有效信息，我还是没法判断  😂

> 看截图似乎在锤子便签中没有生效适配规则，你可以看看是否像下图一样显示正确的适配规则

![image](https://user-images.githubusercontent.com/81074/139529479-d49efef1-4e23-4619-b6e8-23a44849a75b.png)

**Reply by Kenshin** [reply]
@KYBIN-9 

请不要使用 Email  回复我，格式会乱掉，请在 Github 回复我就好。（乱倒的内容我删除了。）

***

> 您好，我刚又摸索一遍，好像是站点适配的问题，我在站点集市安装那个锤子适配好几次了，还是显示0条。

至于你说的这个问题，是否安装了同步助手？如果是的话，无法安装站点集市应该是个 Bug，解决方案可以参考 https://github.com/Kenshin/simpread/discussions/2342 提供的思路（只是把插件改为站点集市，步骤都是一样的。

PS：这个 Bug 只是个别情况，因为我这边一直无法重现，所以没有太好的解决方案。

**Reply by YB-Kang** [reply]
我把刚才那条回复删除重新写，并重新录制视频。
---
https://user-images.githubusercontent.com/55050981/139576106-a6bbfcf4-7e00-416a-8c29-3ac65725b41c.mp4

我根据 #2342 的方法解决了站点集市适配问题，锤子插件选项也能打开了。
但是进入阅读模式后总是显示正文获取异常。手动框选没办法获取到正文。

另外，我录制的视频最后是一篇百度新闻，显示智能感知正文失败，但可以手动框选进入。这个问题有什么问题吗，我用了这么久简悦基本的网页都能自动适配到的。

**Reply by YB-Kang** [reply]
附上百度新闻链接  https://mbd.baidu.com/newspage/data/landingsuper?context=%7B%22nid%22%3A%22news_9110397156903103021%22%7D&n_type=1&p_from=3

**Reply by Kenshin** [reply]
> 但是进入阅读模式后总是显示正文获取异常

嗨，无需理会这个提示，因为你的正文内容比较少，所以才有这个提示。你可以在锤子便签中增加一些内容，随便找点什么复制过去，然后再测试，就可以了。

PS：视频显示的正文明显是获取成功后的内容。

**Reply by Kenshin** [reply]
> 附上百度新闻链接 https://mbd.baidu.com/newspage/data/landingsuper?context=%7B%22nid%22%3A%22news_9110397156903103021%22%7D&n_type=1&p_from=3

至于这个问题是因为解析失败造成的，也算是正常情况。
另外，能否多找一些百度新闻的链接给我，我可以考虑适配它。

**Reply by YB-Kang** [reply]
> > 但是进入阅读模式后总是显示正文获取异常
> 
> 嗨，无需理会这个提示，因为你的正文内容比较少，所以才有这个提示。你可以在锤子便签中增加一些内容，随便找点什么复制过去，然后再测试，就可以了。
> 
> PS：视频显示的正文明显是获取成功后的内容。

这个问题已解决  谢谢

**Reply by YB-Kang** [reply]
> 至于这个问题是因为解析失败造成的，也算是正常情况。 另外，能否多找一些百度新闻的链接给我，我可以考虑适配它。

发到您邮箱了，4个mdb.baidu.com开头，2个百家号网页

**Reply by Kenshin** [reply]
@KYBIN-9 

已收到~
